scipy/stats/_axis_nan_policy.py

```diff
@@ -88,6 +88,68 @@ def _remove_nans(samples, paired):
     return [sample[not_nans] for sample in samples]
 
 
+def _remove_sentinel(samples, paired, sentinel):
+    "Remove sentinel values from paired or unpaired 1D samples"
+    # could consolidate with `_remove_nans`, but it's not quite as simple as
+    # passing `sentinel=np.nan` because `(np.nan == np.nan) is False`
+
+    # potential optimization: don't copy arrays that don't contain sentinel
+    if not paired:
+        return [sample[sample != sentinel] for sample in samples]
+
+    # for paired samples, we need to remove the whole pair when any part
+    # has a nan
+    sentinels = (samples[0] == sentinel)
+    for sample in samples[1:]:
+        sentinels = sentinels | (sample == sentinel)
+    not_sentinels = ~sentinels
+    return [sample[not_sentinels] for sample in samples]
+
+
+def _masked_arrays_2_sentinel_arrays(samples):
+    # masked arrays in `samples` are converted to regular arrays, and values
+    # corresponding with masked elements ar replaced with a sentinel value
+
+    # return without modifying arrays if none have a mask
+    has_mask = False
+    for sample in samples:
+        mask = getattr(sample, 'mask', False)
+        has_mask = has_mask or np.any(mask)
+    if not has_mask:
+        return samples, None  # None means there is no sentinel value
+
+    # Choose a sentinel value. We can't use `np.nan`, because sentinel (masked)
+    # values are always omitted, but there are different nan policies.
+    for i in range(len(samples)):
+        # Things get more complicated if the arrays are of different types.
+        # We could have different sentinel values for each array, but
+        # the purpose of this code is convenience, not efficiency.
+        samples[i] = samples[i].astype(np.float64, copy=False)
+
+    max_possible, eps = np.finfo(np.float64).max, np.finfo(np.float64).eps
+
+    sentinel = max_possible
+    while sentinel > 0:
+        for sample in samples:
+            if np.any(sample == sentinel):
+                sentinel *= (1 - 2*eps)  # choose a new sentinel value
+                break
+        else: # when sentinel value is OK, break the while loop
```

Comment:
```suggestion
        else:  # when sentinel value is OK, break the while loop
```
PEP8: `scipy/stats/_axis_nan_policy.py:137:14: E261 at least two spaces before inline comment`